main.go

```diff
@@ -572,7 +572,9 @@ func main() {
 	if *flUsername == "" {
 		// username and user@host URLs are validated as mutually exclusive
 		if u, err := url.Parse(*flRepo); err == nil { // it may not even parse as a URL, that's OK
-			if u.User != nil && u.Scheme != "ssh" {
+			// Note that `ssh://user@host/path` URLs need to retain the user
+			// field. Out of caution, we only handle HTTP(S) URLs here.
+			if u.User != nil && (u.Scheme == "http" || u.Scheme == "https") {
```

Comment:
Why not keep test on schema ssh here ?
Note that for url  format `git@github.com:kubernetes/git-sync` method `url.Parse` return an err.

Reply:
I don't know what OTHER schemes might have the problem that `ssh://...` has, but I *do* know that this logic was only intended for HTTP and HTTPS.  I'd rather fall back on not changing the URL than changing it wrong (as I did with SSH)